tsdb/block.go

```diff
@@ -656,7 +656,7 @@ func (pb *Block) CleanTombstones(dest string, c Compactor) (*ulid.ULID, bool, er
 	}
 
 	meta := pb.Meta()
-	uid, err := c.Write(dest, pb, pb.meta.MinTime, pb.meta.MaxTime, &meta)
+	uid, err := c.Write(dest, pb, pb.meta.MinTime, pb.meta.MaxTime, false, &meta)
```

Comment:
The problem is the meta file gets created inside the `Write` method, so we still need to pass something like `ooo` into that method. If we want to change the out-of-order signal from a hint to a dedicated field, maybe we could do that in a separate PR?

Reply:
Nill gets passed here https://github.com/prometheus/prometheus/blob/501bc6419ee54ee5617d75c3bc16ecf36d94c552/tsdb/db.go#L1306

Instead, since we know we are compacting the OOO had we know we might want to pass a parent meta instantiated with OOO = true

And then if parent is not nil here we could set parents ooo value https://github.com/prometheus/prometheus/blob/7852a7c5162a7ebd21a8766e3a75d7de1ac1cf3c/tsdb/compact.go#L548-L552

Reply:
So something like this?

```
	if parent != nil {
		meta.Compaction.Parents = []BlockDesc{
			{ULID: parent.ULID, MinTime: parent.MinTime, MaxTime: parent.MaxTime},
		}
		if parent.Compaction.FromOutOfOrder() {
			meta.Compaction.SetOutOfOrder()
		}
	}
```

Reply:
That make sense. The parent meta would be discarded, but that doesn't seem like a big deal. I wasn't sure if `parent` had some special meaning, but you think it's fine to use in that way?

Reply:
I think its fine. Maybe we could rename parent to `base`. 

The purpose of this parent meta is for lvl 2+ compacted blocks to have a history of which blocks conform it which is something that is not natural for prometheus since prometheus does not compact above lvl 1. That is a feature other projects like thanos or mimir use more often but prometheus was ok with including in the code for convenience.

I think we are fine extending this slightly and calling it base meta so the resulting meta can have all the info the users may want. And in that context knowing that it was an OOO block is actually useful for prometheus itself.